libs/ledger-live-common/src/families/celo/js-buildTransaction.ts

```diff
@@ -2,23 +2,118 @@ import type { Transaction } from "./types";
 import type { Account } from "../../types";
 import { CeloTx } from "@celo/connect";
 import { celoKit } from "./api/sdk";
+import { BigNumber } from "bignumber.js";
+import { getVote } from "./logic";
 
+// TODO: a lot of this code overlaps with getFeesForTransaction, but not all. Check if passing an extracted
+// celoTransaction from here to getFees estimateGas would work
```

Comment:
here's a lot of code that I want to refactor, I haven't done it yet because I wanted to have all modes of transactions ready. A lot of code is similar but not the same so I'll need to refactor carefully, ideally by writing bot specs first